apiserver/facades/client/charmhub/charmhub.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmhub
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/names/v4"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/charmhub"
+)
+
+var logger = loggo.GetLogger("juju.apiserver.charmhub")
+
+type Client interface {
+	Info(ctx context.Context, name string) (charmhub.InfoResponse, error)
+}
+
+// API provides the charmhub API facade for version 1.
+type CharmHubAPI struct {
+	auth facade.Authorizer
+
+	//newClientFunc is for testing purposes to facilitate using mocks.
+	newClientFunc func(charmhub.Config) (Client, error)
+}
+
+func NewFacade(ctx facade.Context) (*CharmHubAPI, error) {
+	auth := ctx.Auth()
+	ncFunc := func(p charmhub.Config) (Client, error) {
+		return charmhub.NewClient(p)
+	}
+	return newCharmHubAPI(auth, ncFunc)
+}
+
+func newCharmHubAPI(authorizer facade.Authorizer, ncFunc func(charmhub.Config) (Client, error)) (*CharmHubAPI, error) {
+	if !authorizer.AuthClient() {
+		return nil, common.ErrPerm
+	}
+	return &CharmHubAPI{auth: authorizer, newClientFunc: ncFunc}, nil
+}
+
+func (api *CharmHubAPI) Info(arg params.Entity) (params.CharmHubCharmInfoResult, error) {
+	logger.Tracef("Info()")
+	tag, err := names.ParseApplicationTag(arg.Tag)
+	if err != nil {
+		return params.CharmHubCharmInfoResult{}, errors.BadRequestf("arg value is empty")
+	}
+	// TODO: (hml) 2020-06-17
+	// Add model config value for charmhub-url to charmhub client New().
+	// once implemented.
+	chClient, err := api.newClientFunc(charmhub.CharmhubConfig())
+	if err != nil {
+		return params.CharmHubCharmInfoResult{}, errors.Annotate(err, "could not get charm hub client")
+	}
+	// TODO:
+	// Create a proper context to be used here.
+	info, err := chClient.Info(context.TODO(), tag.Id())
```

Comment:
Yeah, it would be good to allow juju to push a context all the way through the stack.